From 78939d18a732612af1a1f769844b6af0647f6de0 Mon Sep 17 00:00:00 2001
From: Wang Quanyang <quanyang.wang@windriver.com>
Date: Wed, 21 Jun 2017 10:23:50 +0800
Subject: [PATCH] arm64: kdump: fix variables be used uninitialized compile
 warning

This patch fix compile warning as below:

kernel-source/include/linux/kexec.h:29:0,
                 from kernel-source/kernel/kexec_core.c: In function '__crash_kexec':
kernel-source/arch/arm64/include/asm/kexec.h:48:3: warning: 'tmp1' may be used uninitialized in this function [-Wmaybe-uninitialized]
   __asm__ __volatile__ (
   ^~~~~~~
kernel-source/arch/arm64/include/asm/kexec.h:46:7: note: 'tmp1' was declared here
   u64 tmp1, tmp2;
       ^~~~
kernel-source/arch/arm64/include/asm/kexec.h:48:3: warning: 'tmp2' may be used uninitialized in this function [-Wmaybe-uninitialized]
   __asm__ __volatile__ (
   ^~~~~~~
kernel-source/arch/arm64/include/asm/kexec.h:46:13: note: 'tmp2' was declared here
   u64 tmp1, tmp2;
             ^~~~

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm64/include/asm/kexec.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/arm64/include/asm/kexec.h b/arch/arm64/include/asm/kexec.h
index b5168e8..0494c17 100644
--- a/arch/arm64/include/asm/kexec.h
+++ b/arch/arm64/include/asm/kexec.h
@@ -43,7 +43,8 @@ static inline void crash_setup_regs(struct pt_regs *newregs,
 	if (oldregs) {
 		memcpy(newregs, oldregs, sizeof(*newregs));
 	} else {
-		u64 tmp1, tmp2;
+		u64 tmp1 = 0;
+		u64 tmp2 = 0;
 
 		__asm__ __volatile__ (
 			"stp	 x0,   x1, [%2, #16 *  0]\n"
-- 
1.7.5.4

